integr8ly/integreatly-operator
# [WIP] Use the cloud resource operator to provision AWS resources

## Description 
Use the [Cloud Resource Operator](https://github.com/integr8ly/cloud-resource-operator/) to provision databases and resources in AWS and point to them in the 3scale and codeready reconcilers. 

This is done by adding a `useExternalResources` key to the Installation CR spec. If this flag is set to true, in Codeready this:
- Creates an RDS instance and configure the che cluster to point to it

In 3scale: 
- Deploys two elasticache instances for `system-redis` and `backend-redis` 
- Deploys an RDS instance for `system-database`
- Creates an SMTP Credential set 
- Creates an S3 bucket 
- Configures 3scale to use these resources 

## Verification
- Run the integreatly operator from this branch 
- Run the cloud resource operator
- Create a new `installation` resource:
```yml
apiVersion: integreatly.org/v1alpha1
kind: Installation
metadata:
  name: example-installation
spec:
  type: managed
  namespacePrefix: integreatly-
  routingSubdomain: example.com
  masterUrl: http://master.example.com
  selfSignedCerts: true
  useExternalResources: true
```
- Create a VPC peering connection and update the two VPC route tables to point to the peering connection 
- Verify that all 3scale pods are running and all codeready pods are running
- Verify that two elasticache instances were created and are available and two RDS instances were created and are available 
- Verify that normal 3scale actions can be performed (inviting a user, creating a service, etc)
